model/task/task.go

```diff
@@ -2741,6 +2742,37 @@ func GetTasksByVersion(versionID string, sortBy []TasksSortOrder, statuses []str
 					tempParentKey: []interface{}{},
 				},
 			},
+			// get any annotation that has at least one issue
+			{
+				"$lookup": bson.M{
+					"from": annotations.Collection,
+					"let":  bson.M{"task_annotation_id": "$task_id", "task_annotation_execution": "$execution"},
+					"pipeline": []bson.M{
+						{
+							"$match": bson.M{
+								"$expr": bson.M{
+									"$and": []bson.M{
+										{
+											"$eq": []string{"$task_id", "$$task_annotation_id"},
+										},
+										{
+											"$eq": []string{"$execution", "$$task_annotation_execution"},
+										},
+										{"$or": []bson.M{
+											{
+												annotations.IssuesKey: bson.M{"$gt": []interface{}{"$" + annotations.IssuesKey, 0}},
+											},
+
+											{
+												annotations.IssuesKey: bson.M{"$ne": []interface{}{"$" + annotations.IssuesKey, bson.TypeNull}},
```

Comment:
Doesn't this or case mean that if Issues is an empty list we'll still return it?

Reply:
Thanks for catching this! I probably need $nor instead.